flow/designs/asap7/mock-array/macro-placement.tcl

```diff
@@ -1,16 +1,22 @@
 # fish out values from single source of truth: config.py
-set vals [regexp -all -inline {\S+} [exec sh -c {cd designs/asap7/mock-array && python3 -c "import config;print(f'{config.rows} {config.cols} {config.pitch_x} {config.pitch_y} {config.array_offset_x} {config.array_offset_y} {config.placement_grid_x} {config.placement_grid_y}')"}]]
-lassign $vals rows cols pitch_x pitch_y array_offset_x array_offset_y placement_grid_x placement_grid_y
+set vals [regexp -all -inline {\S+} [exec sh -c {cd designs/asap7/mock-array && python3 -c "import config;print(f'{config.rows} {config.cols} {config.space_x} {config.space_y} {config.margin_x} {config.margin_y} {config.placement_grid_x} {config.placement_grid_y}')"}]]
+lassign $vals rows cols space_x space_y margin_x margin_y placement_grid_x placement_grid_y
 
 set block [ord::get_db_block]
 set units [$block getDefUnits]
 
+# place Element in (rows, cols)
+# Element spacing to the right is calculated by space_x + it's width
+# Element spacing to above is calculated by space_y + it's height
 for {set row 0} {$row < $rows} {incr row} {
   for {set col 0} {$col < $cols} {incr col} {
     set inst [$block findInst [format "ces_%d_%d" $row $col]]
+    set bbox [$inst getBBox]
+    set w [$bbox getDX]
+    set h [$bbox getDY]
 
-    set x [expr int((($array_offset_x + (($placement_grid_x * $pitch_x) * $col)) * $units))]
-    set y [expr int((($array_offset_y + (($placement_grid_y * $pitch_y) * $row)) * $units))]
+    set x [expr int((($margin_x + (($placement_grid_x * $space_x) * ($col + 1.0))) * $units) + ($w * $col))]
```

Comment:
i cant figure out what is going on here reviewing on my phone :-)

Reply:
changing to space spec require calculation from size of Element. This allow 0 spacing between Element in either (or both) direction.